playback/src/audio_backend/mod.rs

```diff
@@ -8,6 +8,9 @@ pub trait Sink {
     fn start(&mut self) -> io::Result<()>;
     fn stop(&mut self) -> io::Result<()>;
     fn write(&mut self, data: &[i16]) -> io::Result<()>;
+    fn write_bytes(&mut self, _data: &[u8]) -> io::Result<()> {
+        Ok(())
+    }
```

Comment:
I agree but then I would have to modify every existing backend which I would assume was not desirable, that's the reason I ended up with this. I did not want that PR to be a virus spreading across code.

Reply:
Would it be an option to add an optional type parameter to `Sink`? Like
```rust
pub trait Sink<S = i16> {
    fn write(&mut self, data: &[S]) -> io::Result<()>; 
}
```

Reply:
> I would have to modify every existing backend which I would assume was not desirable

Modifying every backend is perfectly fine if the overall result is nicer.

-----

@Johannesd3 I'm not sure this actually works. The player holds a `Box<dyn Sink>`, so it is important that all sinks expose the same interface.

Reply:
It works for certain, but not without modifying very much existing code, so probably it was a bad idea.

Reply:
> > I would have to modify every existing backend which I would assume was not desirable
> 
> Modifying every backend is perfectly fine if the overall result is nicer.
> 
So... are you really sure you want me to do that?

Reply:
I think it would be cleaner. Matching from each backend doesn't really add any code complexity, and it makes more sense than multiple methods to read/write the same underlying data, albeit in a different format.

Reply:
done